### PR TITLE
improve javascript

### PR DESCRIPTION
**Pull request by marcobiedermann**
Improve JavaScript:
- Cache `window` and `document` object
- Hoist variables in `for` statement to top of function
- Replace space indentation with tabs
- Improve coding style

### CODE REVIEW

File: src/superplaceholder.js [hunk 1]
```diff
@@ -52,43 +58,50 @@
...
 			// character of the string.
 			self.el.setAttribute('placeholder', str.substr(0, index + 1) + (index === str.length - 1 || !self.options.showCursor ? '' : self.options.cursor));
+
 			if (index === str.length - 1) {
 				callback();
 			}
 		}
-		for (var i = 0; i < str.length; i++) {
+
+		for (i = 0, length = str.length; i < length; i++) {
```
Comment (chinchang) [inline src/superplaceholder.js:96]: Caching length is unnecessary now-a-days with modern browsers doing it already internally

**Comment by chinchang** [comment]
@marcobiedermann Looks good. Please remove the `dist/*` files from this PR. They will be auto generated when PR is merged into master.

File: src/superplaceholder.js [hunk 2]
```diff
@@ -99,17 +112,20 @@
 	};
 
 	var superplaceholder = function (params) {
-		if (!isPlaceHolderSupported) { return; }
+		if (!isPlaceHolderSupported) {
```
Comment (chinchang) [inline src/superplaceholder.js]: IMO, flow breaking statements (like return, continue etc) can be left on the same line.

**Comment by marcobiedermann** [comment]
@chinchang I've updated my pull request to your suggested changes!

**Comment by marcobiedermann** [comment]
@chinchang BTW, how do you auto generate the files in `dist`? Do you use some git hooks or any third party service?

**Comment by chinchang** [comment]
@marcobiedermann I think there is confusion...the dist/ files were to be removed from your PR, but I think you removed them from the repo itself :) Just that your [PR changes](https://github.com/chinchang/superplaceholder.js/pull/6/files#diff-cd151fedcebfe437914694db67e4c5d0) should not show any change regarding dist/ files

**Comment by chinchang** [comment]
Right now its a local script that does the tagging, building etc stuff whenever a release is ready on master. But I suppose I could do that with travis as well.

**Comment by marcobiedermann** [comment]
Ping @chinchang 😄

**Comment by marcobiedermann** [comment]
I'm closing this issue because this project is no longer being maintained

**Pull request closed without merging**